pkg/cluster/internal/haproxy/config.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package haproxy
+
+import (
+	"bytes"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+// ConfigData is supplied to the haproxy config template
+type ConfigData struct {
+	ControlPlanePort int
+	AdminPort        int
+	BackendServers   map[string]string
+}
+
+// DefaultConfigTemplate is the haproxy config template
+const DefaultConfigTemplate = `
```

Comment:
@munnerz @neolit123 are you familiar with haproxy config? I need to do some reading :-)

Reply:
Correct, not looking to block on that! If anyone else is already familiar a review would be appreciated, otherwise I intend to do some reading for the future :-)

Re: admin -- I think we should remove it initially and deal with the implications in a follow-up later. I've been thinking more about the config for these things - have a couple ideas I want to run by you all soon but I don't think they need to be in this PR.

Reply:
Ok, admin console is gone now!